basyx.components/basyx.components.docker/basyx.components.AASServer/src/main/java/org/eclipse/basyx/components/aas/mongodb/MongoDBAASAggregator.java

```diff
@@ -456,13 +445,12 @@ public void reset() {
 		aasProviderMap.clear();
 	}
 
-	@SuppressWarnings("deprecation")
 	private void init() {
 		List<AssetAdministrationShell> data = mongoOps.findAll(AssetAdministrationShell.class, aasCollection);
 		for (AssetAdministrationShell aas : data) {
 			String aasId = aas.getIdentification().getId();
 			logger.info("Adding AAS from DB: " + aasId);
-			MongoDBAASAPI aasApi = new MongoDBAASAPI(config, aasId);
+			MongoDBAASAPI aasApi = new MongoDBAASAPI(config, aasId, mongoClient);
 			MultiSubmodelProvider provider = createMultiSubmodelProvider(aasApi);
 			addSubmodelsFromDB(provider, aas);
 			aasProviderMap.put(aas.getIdentification().getId(), provider);
```

Comment:
Is there a way to close the connection after the work is done?

Reply:
Since the MongoDBAASAPI may modify the AAS in the DB, it needs an active connection. Thus, in my opinion the connection needs to stay open here